client/client_test.go

```diff
@@ -33,6 +34,91 @@ func TestAddGetLeaf(t *testing.T) {
 	// TODO: Build a GetLeaf method and test a full get/set cycle.
 }
 
+func TestGetByIndex(t *testing.T) {
+	env, err := integration.NewLogEnv(context.Background(), 0, "TestGetByIndex")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	logID, err := env.CreateLog()
+	if err != nil {
+		t.Fatalf("Failed to create log: %v", err)
+	}
+
+	cli := trillian.NewTrillianLogClient(env.ClientConn)
+	client := New(logID, cli, testonly.Hasher, env.PublicKey)
+	// Add a few test leaves.
+	leafData := [][]byte{
+		[]byte("A"),
+		[]byte("B"),
+		[]byte("C"),
+	}
+
+	// TODO(gdbelvin): Replace with batch API.
+	// TODO(gdbelvin): Replace with AddSequenedLeaves API.
+	for _, l := range leafData {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
+		defer cancel()
+		if err, want := client.AddLeaf(ctx, l), codes.DeadlineExceeded; grpc.Code(err) != want {
```

Comment:
I wish we could do this without the timeouts. Have you considered having a version of AddLeaf that doesn't block?

A more general question: some logs take a long time to integrate leaves (MMDs go up to 24h). Is the AddLeaf behavior desirable considering such scenarios? Should we dissociate adding from verifying? (I don't expect any changes related to this, it's just a question of mine.)